sql/hive/src/test/scala/org/apache/spark/sql/sources/JsonHadoopFsRelationSuite.scala

```diff
@@ -32,10 +32,6 @@ class JsonHadoopFsRelationSuite extends HadoopFsRelationTest {
   override protected def supportsDataType(dataType: DataType): Boolean = dataType match {
     case _: NullType => false
     case _: BinaryType => false
-    // `TimestampType` is disabled because `DatatypeConverter.parseDateTime()`
-    // in `DateTimeUtils` parses the formatted string wrongly when the date is
-    // too early. (e.g. "1600-07-13T08:36:32.847").
-    case _: TimestampType => false
```

Comment:
It should have been enabled in the previous PR. My initial proposal was just to change the write path where both CSV and JSON used we used `DatatypeConverter.parseDateTime` for read path.

However, the previous PR ended up with changing read path as well switching `DatatypeConverter.parseDateTime` to `FastDateFormat` not having this problem as described above.